Changelog.md

```diff
@@ -17,6 +17,7 @@ Bugfixes:
  * Control Flow Graph: Perform proper virtual lookup for modifiers for uninitialized variable and unreachable code analysis.
  * Immutables: Fix wrong error when the constructor of a base contract uses ``return`` and the parent contract contains immutable variables.
  * IR Generator: Fix IR syntax error when copying storage arrays of structs containing functions.
+ * IR Generator: Fix IR syntax error when copying reference types in calldata and storage to struct members in memory.
```

Comment:
It's not only struct members, but also array members, isn't it?
Might actually be worth another simple test case.

Reply:
Actually yeah, also array members, the test `test/libsolidity/semanticTests/structs/copy_from_calldata.sol` shows it.

Reply:
We could add a few more tests, i.e. copying storage arrays to a memory array element, etc.... we can also consider that part of https://github.com/ethereum/solidity/issues/12559, but we can also do it while we're at it...